lib/chromic_pdf/pdf/browser/channel.ex

```diff
@@ -50,34 +47,117 @@ defmodule ChromicPDF.Browser.Channel do
   def init(args) do
     {:ok, conn} = Connection.start_link(self(), args)
 
-    dispatch = fn call ->
-      Connection.dispatch_call(conn, call)
-    end
-
-    {:ok, %__MODULE__{dispatch: dispatch, protocols: []}}
+    {:ok, %{conn: conn, waitlist: []}}
   end
 
-  @impl GenServer
   # Starts protocol processing, asynchronously sends result message when done.
-  def handle_call(
-        {:run_protocol, protocol},
-        from,
-        %__MODULE__{dispatch: dispatch, protocols: protocols} = state
-      ) do
-    protocol = Protocol.init(protocol, &GenServer.reply(from, &1), dispatch)
-
-    {:noreply, update_protocols(state, [protocol | protocols])}
+  @impl GenServer
+  def handle_call({:run_protocol, protocol, ref}, from, state) do
+    task = %{protocol: protocol, from: from, ref: ref}
```

Comment:
Kinda liked the `ChannelTask` 🤷 but also fine to keep it as a private thingy in the channel